compiler/luci/pass/src/FoldShapePass.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FoldShapePass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+namespace
+{
+
+template <loco::DataType OutType> luci::CircleConst *shape_of(luci::CircleNode *input_tensor)
+{
+  auto name = input_tensor->name();
+  auto shape_status = input_tensor->shape_status();
+  auto rank = input_tensor->rank();
+  assert(name.length() > 0);
+  assert(shape_status == luci::ShapeStatus::VALID);
+  assert(rank > 0);
```

Comment:
These would have to be implemented as if-statements.

Reply:
Is this condition check related with the above case which should reject conversion?

Reply:
I mean that if these conditions are not satisfied, I should reject the conversion? (Not just assert)

Reply:
Yes :) `assert` is used for illegal inputs. Users can give those kinds of inputs, which means they are legal.